dash/development/_jl_components_generation.py

```diff
@@ -0,0 +1,523 @@
+                name=prop_name,
+                type=jl_type_name,
+                description=description,
+                is_required="required" if required else "optional",
+            )
+        )
+    return "{indent_spacing}- `{name}` ({type}{is_required}){description}".format(
+        indent_spacing=indent_spacing,
+        name=prop_name,
+        type="{}; ".format(jl_type_name) if jl_type_name else "",
+        description=(": {}".format(description) if description != "" else ""),
+        is_required="required" if required else "optional",
+    )
+
+
+# this logic will permit passing blank Julia prefixes to
+# dash-generate-components, while also enforcing
+# lower case names for the resulting functions; if a prefix
+# is supplied, leave it as-is
+def format_fn_name(prefix, name):
+    if prefix:
+        return "{}_{}".format(prefix, name.lower())
+    return name.lower()
+
+
+def generate_metadata_strings(resources, metatype):
+    def noting_or_string(v):
+        return '"{}"'.format(v) if v else "nothing"
+    return [jl_resource_tuple_string.format(
+        relative_package_path=noting_or_string(resource.get("relative_package_path", "")),
+        external_url=noting_or_string(resource.get("external_url", "")),
+        dynamic=str(resource.get("dynamic", 'nothing')).lower(),
+        type=metatype,
+        async_string=":{}".format(str(resource.get("async")).lower())
+        if "async" in resource.keys()
+        else 'nothing'
+    ) for resource in resources]
+
+
+def generate_package_file(project_shortname, components, pkg_data, prefix):
+    package_name = jl_package_name(project_shortname)
+
+    sys.path.insert(0, os.getcwd())
+    mod = importlib.import_module(project_shortname)
+    js_dist = getattr(mod, "_js_dist", [])
+    css_dist = getattr(mod, "_css_dist", [])
+    project_ver = pkg_data.get("version")
+
+    resources_dist = ",\n".join(
+        generate_metadata_strings(js_dist, "js") + generate_metadata_strings(css_dist, "css")
+    )
+
+    package_string = jl_package_file_string.format(
+        package_name=package_name,
+        component_includes="\n".join(
+            [jl_component_include_string.format(name=format_fn_name(prefix, comp_name)) for comp_name in components]
+        ),
+        resources_dist=resources_dist,
+        version=project_ver,
+        project_shortname=project_shortname
+
+    )
+    file_path = os.path.join("src", package_name + ".jl")
+    with open(file_path, "w") as f:
+        f.write(package_string)
+    print("Generated {}".format(file_path))
+
+
+def generate_toml_file(project_shortname, pkg_data):
+    package_author = pkg_data.get("author", "")
+    project_ver = pkg_data.get("version")
+    package_name = jl_package_name(project_shortname)
+    u = uuid.UUID(jl_dash_uuid)
+    package_uuid = uuid.UUID(hex=u.hex[:-12] + hex(hash(package_name))[-12:])
+
+    authors_string = 'authors = ["{}"]'.format(package_author) if package_author else ""
```

Comment:
minor: with no authors we'll end up with a funny-looking blank line in the output

Reply:
You could simply swap these two lines so the blank line falls between sections:

https://github.com/plotly/dash/pull/1197/files#diff-4de66a02832ac93535450311a2298fdeR87-R88

Reply:
or just move one of the newline characters from `jl_projecttoml_string ` to `authors_string`

Reply:
@alexcjohnson nice catch! fixed in 2e813bf1

Reply:
As @mbauman mentioned, we should also update this in `any`